### DISCUSSION TITLE
DataGrid Vertical Scroll Bar Problem

### DISCUSSION THREAD

**Discussion by keynes1981** [AvaloniaUI/Avalonia#17839] [Q&A]
I found a strange problem when I was using DataGrid: 
1、When the app is started, the window size is normal
2、Added enough data into DataGrid to make the Vertical Scroll Bar be shown.
2、Use the mouse wheel to scroll   the Vertical Scroll Bar down to the bottom of DataGrid.
3、Maxiumize the window size
4、Scroll the Vertical Scroll Bar up to the top of DataGrid
5、Scroll  the Vertical Scroll Bar down  until  it cannot move, the last item of the datagrid is already in view, but the the scroll bar doesn't reach the bottom,it has  a few distance to the bottom.

![1111](https://github.com/user-attachments/assets/f3f184ef-4b02-4f6e-be5d-29a2222fad28)
![无标题](https://github.com/user-attachments/assets/767e2b32-b6d6-4954-85a0-b07166d32609)
![无标题1](https://github.com/user-attachments/assets/5bed14d6-2016-47ef-9a04-4b396f340dcb)
[AvaloniaApplication4.zip](https://github.com/user-attachments/files/18259475/AvaloniaApplication4.zip)

`<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication5.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        x:Class="AvaloniaApplication5.Views.MainWindow"
        x:DataType="vm:MainWindowViewModel"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication5">

	<DataGrid Margin="50" ItemsSource="{Binding People}"
			   GridLinesVisibility="All"
			   BorderThickness="1" BorderBrush="Gray">
		<DataGrid.Styles>
			<Style Selector="DataGrid /template/ScrollBar#PART_VerticalScrollbar">
				<Setter Property="AllowAutoHide" Value="False"/>
			</Style>
		</DataGrid.Styles>
		<DataGrid.Columns>
			<DataGridTextColumn Header="First Name"  Binding="{Binding FirstName}"/>
			<DataGridTextColumn Header="Last Name" Binding="{Binding LastName}" />
		</DataGrid.Columns>
	</DataGrid>

</Window>`
`using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace AvaloniaApplication5.ViewModels
{
    public partial class MainWindowViewModel : ViewModelBase
    {
        public ObservableCollection<Person> People { get; set; }
        public MainWindowViewModel()
        {
            List<Person> persons = [];
            for (var i = 0; i < 50; i++)
            {
                var person = new Person($"First Name {i}", $"Last Name {i}");
                persons.Add(person);
            }

            People = new ObservableCollection<Person>(persons);
        }
    }

    public class Person(string firstName, string lastName)
    {
        public string FirstName { get; set; } = firstName;
        public string LastName { get; set; } = lastName;
    }
}`

**Comment by msneijders** [comment]
Avalonia's ControlCatalog.Desktop DataGridPage shows the same vertical scrollbar behavior. I think you should turn this into an issue, to see if it can be fixed.

**Reply by fallenleavesguy** [reply]
me too. but I don't know how to fix it

**Reply by lextm** [reply]
@fallenleavesguy This fork almost redesigned the internals and invented a much improved scrolling mechanism, so you might want to switch to it and see the changes

https://github.com/wieslawsoltes/ProDataGrid
